### PR TITLE
test: No warnings

### LINKED ISSUE #2: elementwise comparison deprecation warnings in unit tests
```
============================================================= warnings summary =============================================================
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-time]
  /Users/swast/src/github.com/googleapis/python-db-dtypes-pandas/tests/unit/test_dtypes.py:142: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    "==": (lambda a, b: a == b),

tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left0-==-right0-expected0-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-date]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-time]
tests/unit/test_dtypes.py::test_timearray_comparisons[left1-==-right1-expected1-time]
  /Users/swast/src/github.com/googleapis/python-db-dtypes-pandas/tests/unit/test_dtypes.py:147: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    "==": (lambda a, b: a != b),

tests/unit/test_dtypes.py: 16 warnings
  /Users/swast/src/github.com/googleapis/python-db-dtypes-pandas/.nox/unit-3-7/lib/python3.7/site-packages/numpy/core/numeric.py:2446: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    return bool(asarray(a1 == a2).all())

tests/unit/test_dtypes.py::test_min_max_median[date]
tests/unit/test_dtypes.py::test_min_max_median[time]
  /Users/swast/src/github.com/googleapis/python-db-dtypes-pandas/.nox/unit-3-7/lib/python3.7/site-packages/numpy/lib/nanfunctions.py:1117: RuntimeWarning: Mean of empty slice
    return np.nanmean(a, axis, out=out, keepdims=keepdims)

```

### PR DESCRIPTION
**Pull request by jimfulton**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-db-dtypes-pandas/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #2 🦕

This builds-on/includes #8, which builds-on/includes #7.

### CODE REVIEW

**Comment by jimfulton** [comment]
Note there are some commits to address comments made on https://github.com/googleapis/python-bigquery/pull/972 :)

**Review by plamut — APPROVED** [review]
LGTM, the comments from https://github.com/googleapis/python-bigquery/pull/972 have been addressed.

Thanks for this!

**Comment by tswast** [comment]
Coverage failure to be addressed in https://github.com/googleapis/python-db-dtypes-pandas/pull/11

**Comment by jimfulton** [comment]
> Coverage failure to be addressed in #11

Oops, I addressed it here.

File: db_dtypes/__init__.py [hunk 1]
```diff
@@ -61,15 +65,23 @@ class TimeArray(core.BaseDatetimeArray):
     _npepoch = numpy.datetime64(_epoch)
 
     @classmethod
-    def _datetime(cls, scalar):
+    def _datetime(
+        cls,
+        scalar,
+        match=re.compile(r"\s*(\d+)(?::(\d+)(:\d+(?:[.]\d+)?)?)?\s*$").match,
```
Comment (tswast) [inline db_dtypes/__init__.py]: Let's use named groups for these.

File: db_dtypes/__init__.py [hunk 2]
```diff
@@ -125,12 +142,16 @@ class DateArray(core.BaseDatetimeArray):
     dtype = DateDtype()
 
     @staticmethod
-    def _datetime(scalar):
+    def _datetime(
+        scalar, match=re.compile(r"\s*(\d+)-(\d+)-(\d+)\s*$").match,
```
Comment (tswast) [inline db_dtypes/__init__.py]: Same here.

File: tests/unit/test_dtypes.py [hunk 3]
```diff
@@ -620,3 +622,61 @@ def test_date_sub():
...
+        ("10000-1-1", "year 10000 is out of range"),
+    ],
+)
+def test_bad_date_parsing(value, error):
+    with pytest.raises(ValueError, match=error):
+        _cls("date")([value])
+
+
+@for_date_and_time
+def test_date___arrow__array__(dtype):
```
Comment (tswast) [inline tests/unit/test_dtypes.py:676]: Note to self: remove in #11

**Review by tswast — APPROVED** [review]

**Comment by tswast** [comment]
@googlebot I consent.

**Pull request merged**